src/t8_data/t8_shmem.c

```diff
@@ -60,9 +60,18 @@ t8_shmem_array_is_initialized (const t8_shmem_array_t array)
 }
 #endif
 
-void
+int
 t8_shmem_init (sc_MPI_Comm comm)
 {
+#ifndef T8_ENABLE_MPI
+  // If we do not use MPI, there is nothing to do.
+  // We only have a single process.
+  return 1;
+#endif
+#ifndef SC_ENABLE_MPICOMMSHARED
+  SC_ABORT ("Trying to use shared memory but SC_ENABLE_MPICOMMSHARED is not set. This should not happen if you use MPI "
+            "v.3.0 or higher. Maybe related to https://github.com/DLR-AMR/t8code/pull/1996.");
+#endif
```

Comment:
Is this only the case when MPI is not linked or also when the number of ranks is 1?